packages/vite/src/plugins/dev-ssr-css.ts

```diff
@@ -0,0 +1,23 @@
+import { Plugin } from 'vite'
+import { isCSS } from '../utils'
+
+export function devStyleSSRPlugin (rootDir: string): Plugin {
+  return {
+    name: 'nuxt:dev-style-ssr',
+    apply: 'serve',
+    enforce: 'post',
+    transform (code, id) {
+      if (!isCSS(id) || !code.includes('import.meta.hot')) {
+        return
+      }
+
+      let moduleId = id
+      if (moduleId.startsWith(rootDir)) {
+        moduleId = moduleId.slice(rootDir.length)
+      }
+
+      // When dev `<style>` is injected, remove the `<link>` styles from manifest
+      return code + `\ndocument.querySelector(\`link[href="/_nuxt${moduleId}"]\`).forEach(i=>i.remove())`
```

Comment:
Can we please use `options.app.assetsPath` in place of `/_nuxt`? 

An alternative idea, we can inject hash of moduleId as an attr by renderer and select it here by unique module hash

Reply:
(merging PR to test asap. can do this later)
```suggestion
      // TODO: Use `app.assetsPath` or unique hash
      return code + `\ndocument.querySelector(\`link[href="/_nuxt${moduleId}"]\`).forEach(i=>i.remove())`
```